app/src/main/java/com/github/sdp_begreen/begreen/activities/SharePostActivity.kt

```diff
@@ -0,0 +1,106 @@
+package com.github.sdp_begreen.begreen.activities
+
+import android.graphics.Bitmap
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import android.widget.EditText
+import android.widget.ImageView
+import com.github.sdp_begreen.begreen.R
+import com.github.sdp_begreen.begreen.entities.Post
+
+class SharePostActivity : AppCompatActivity() {
+
+    private lateinit var postImageView : ImageView;
+    private lateinit var postTitleEditText : EditText;
+    private lateinit var sharePostBtn : Button;
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_share_post)
+
+        setupUI()
+    }
+
+    /**
+     * Helper function to setups ui buttons
+     */
+    private fun setupUI() {
+
+        // Bind all UIs
+        postTitleEditText = findViewById(R.id.postTitleEditText);
+        sharePostBtn = findViewById(R.id.sharePostBtn)
+        postImageView = findViewById(R.id.postImageView);
+
+        // Display the post image.
+        displayPostImage()
+
+        // Setup the share button action
+        sharePostBtn.setOnClickListener {
+            sharePost()
+        }
+    }
+
+    /**
+     * Helper function to return the image post which is an extra of the intent
+     */
+    private fun getPostImage() : Bitmap? {
+
+        val image = intent.extras?.get(AddNewPostActivity.EXTRA_IMAGE_BITMAP) as? Bitmap
+
+        return image
+    }
+
+    /**
+     * Helper function to return the post title
+     */
+    private fun getPostTitle() : String {
+        return postTitleEditText.text.toString()
+    }
+
+    /**
+     * Helper function to display the post image
+     */
+    private fun displayPostImage() {
+
+        // If the intent that launched the activity doesn't have the correct extra, "image" will be null.
+        // If so, we finish the activity. Otherwise we can display the image.
+
+        var image = getPostImage()
+
+        if (image != null) {
+            postImageView.setImageBitmap(image)
+        } else {
+            finish()
+        }
```

Comment:
```suggestion
       image?.let { postImageView.setImageBitmap(it) } ?: finish()
```

Reply:
Actually, I prefer your version when we want to assign a value to a variable, but when we have two operations (void) I prefer to keep the old version which is more understandable and clear.